### ISSUE TITLE
`#` incorrectly parsed as start of comment in `IFS=# while loop`

### ISSUE THREAD

**Issue by rafasc** [koalaman/shellcheck#1684]
- Rule Id (false positives): SC1057 SC1073 SC1058 SC1072
-  version: 0.7.0
- [x] The rule's wiki page does not already cover this
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

while IFS=# read -r line _ ; do
    printf '%s\n' "$line"
done
```

#### Here's what shellcheck currently says:
```
In mwe line 3:
while IFS=# read -r line _ ; do
^---^ SC1057: Did you forget the 'do' for this loop?
^-- SC1073: Couldn't parse this while loop. Fix to allow more checks.


In mwe line 5:
done
^-- SC1058: Expected 'do'.
    ^-- SC1072: Unexpected keyword/token. Fix any mentioned problems and try again.

For more information:
  https://www.shellcheck.net/wiki/SC1057 -- Did you forget the 'do' for this ...
  https://www.shellcheck.net/wiki/SC1058 -- Expected 'do'.
  https://www.shellcheck.net/wiki/SC1072 -- Unexpected keyword/token. Fix any...
```

**Comment by koalaman** [comment]
You're right, this is a bug. As a temporary workaround, you can quote it , e.g. `IFS="#"`
